src/Apps/Gene/Routes/GeneShow.tsx

```diff
@@ -26,7 +26,7 @@ export const GeneShow: React.FC<GeneShowProps> = ({ gene }) => {
             {gene.displayName || gene.name}
           </Text>
 
-          <FollowGeneButtonQueryRenderer id={gene.internalID} />
+          <FollowGeneButtonFragmentContainer gene={gene} />
```

Comment:
I'm not sure if this is the right change? You switched away from deferring those calls to requiring that data up-front in order to initially render the page (adding `isFollowed` to the fragment).

Would it be better to keep this as a deferred renderer (that's what we want to do for personalized content, generally) - and simply return early from there if there's no user?

Reply:
I think that sounds more appealing! (and somehow similar to the first implementation)

Reply:
and most of the data(besides isFollowed) is already fetched by the components calling the former `FollowGeneButtonQueryRenderer`.
We talked about having the `isFollowed` on MP as a fallback to always `false` if the user is not logged in to save from hitting the loader when not needed too

Reply:
We want personalized data (like `isFollowed`) to be in its own query renderer. This allows those queries to skip the cache, and any other page-data queries (like gene page, a list of genes, etc.) to be cacheable (even for logged-in).

So keeping follow buttons in their own query renderer _is_ what we want.

Seems like it should be not hard to suppress the call it was making (and MP immediately responds with `isFollowed: false`), but I haven't really checked out the code!

If there's some complexity where an early return _skips rendering the follow button altogether_ - then yea, maybe there needs to be a wrapper that renders _either_ the query renderer component _or_ the 'default' follow button in the un-saved state?

(something like that - I'm being a touch hand-wave-y)

Reply:
I think I'm with you but all the data we query before(slug, internalID) should come from the fragment, now the personalized data totally agree with the separate query.
How about a compromise, we keep the other un-personalized data in the fragment and add a lazyLoad in the same element just to query the personalized data(isFollowed)?

re MP we took a quick look the the field `isFollowed` his a loader there, must be simple to fallback false given unauthenticated users

Reply:
Yep @araujobarret that sounds good! I'm glossing over the slug/id/name part as well - that's of course totally fine to include up-front.